src/main/scala/com/gu/support/workers/model/CheckoutFailureReasons.scala

```diff
@@ -0,0 +1,92 @@
+package com.gu.support.workers.model
+
+object CheckoutFailureReasons {
+
+  val all = List(
+    PaymentMethodDetailsIncorrect,
+    PaymentMethodTemporarilyDeclined,
+    PaymentMethodUnacceptable,
+    PaymentProviderUnavailable,
+    PaymentRecentlyTaken,
+    Unknown
+  )
+
+  def fromString(string: String): Option[CheckoutFailureReason] = all.find(_.asString == string)
+
+  sealed trait CheckoutFailureReason {
+    def asString: String
+  }
+
+  case object PaymentMethodDetailsIncorrect extends CheckoutFailureReason {
```

Comment:
cc @michaelwmcnamara @Mullefa @joelochlann - I would like to make a change to use these same reasons in payment-api responses. 

This will mean that clients don't need to know about all of the different Stripe/PayPal specific errors. It also allows us to have a single function (in support-frontend) which maps a CheckoutFailureReason to a human friendly error message. (This can be shared across all products).

Any thoughts/preferred alternatives to that idea before I start implementing it?

Reply:
We discussed this previously and made an explicit decision to pass the specific Paypal and Stripe errors to the client. But let's catch up to discuss the pros and cons of this approach with a view to possibly changing it.